packaging/sles/ec2-hibernate-linux-agent.spec

```diff
@@ -0,0 +1,146 @@
+   thread to touch all of its blocks to make sure that EBS volumes
+   are pre-warmed.
+3. It updates the offset of the swap file in the kernel using
+   SNAPSHOT_SET_SWAP_AREA ioctl.
+4. It daemonizes and starts a polling thread to listen for instance
+   termination notifications.
+
+%prep
+%autosetup -n %{project}-%{version}
+ 
+%build
+%py3_build
+
+%install
+%py3_install --root %{buildroot}
+
+mkdir -p %{buildroot}%{python3_sitelib}
+mkdir -p "%{buildroot}%{_unitdir}"
+mkdir -p %{buildroot}%{_sysconfdir}/acpi/events 
+mkdir -p %{buildroot}%{_sharedstatedir}/hibinit-agent
+mkdir -p %{buildroot}%{_sysconfdir}/acpi/actions
+mkdir -p %{buildroot}%{_sbindir}
+
+install -p -m 644 "%{_builddir}/%{project}-%{version}/hibinit-agent.service" %{buildroot}%{_unitdir}
+install -p -m 644 "%{_builddir}/%{project}-%{version}/acpid.sleep.conf" %{buildroot}%{_sysconfdir}/acpi/events/sleepconf
+
+mkdir -p %{buildroot}%{_prefix}/lib/systemd/logind.conf.d
+mkdir -p %{buildroot}%{_prefix}/lib/systemd/system-sleep
+
+install -p -m 644 "%{_builddir}/%{project}-%{version}/etc/hibinit-config.cfg" %{buildroot}/%{_sysconfdir}/hibinit-config.cfg
+install -p -m 644 "%{_builddir}/%{project}-%{version}/packaging/rhel/00-hibinit-agent.conf" %{buildroot}%{_prefix}/lib/systemd/logind.conf.d/00-hibinit-agent.conf
+install -p -m 755 "%{_builddir}/%{project}-%{version}/packaging/rhel/acpid.sleep.sh" %{buildroot}%{_sysconfdir}/acpi/actions/sleep.sh
+install -p -m 755 "%{_builddir}/%{project}-%{version}/packaging/rhel/sleep-handler.sh" %{buildroot}%{_prefix}/lib/systemd/system-sleep/sleep-handler.sh
+
+#Disable transparent huge page
+mkdir -p  %{buildroot}%{_sysconfdir}/tuned/nothp_profile
+install -p -m 644 "%{_builddir}/%{project}-%{version}/packaging/rhel/tuned.conf" %{buildroot}%{_sysconfdir}/tuned/nothp_profile/tuned.conf
+
+ln -sf %{_sbindir}/service %{buildroot}%{_sbindir}/rchibinit-agent
+
+%files
+%doc README.md
+%license LICENSE.txt
+%config(noreplace) %{_sysconfdir}/hibinit-config.cfg
+%{_unitdir}/hibinit-agent.service
+%{_bindir}/hibinit-agent
+%{_sbindir}/rchibinit-agent
+%config(noreplace) %{_sysconfdir}/acpi/events/sleepconf
+%config(noreplace) %{_sysconfdir}/acpi/actions/sleep.sh
+%{python3_sitelib}/ec2_hibinit_agent-*.egg-info/
+%dir %{_sharedstatedir}/hibinit-agent
+%ghost %attr(0600,root,root) %{_sharedstatedir}/hibinit-agent/hibernation-enabled
+
+%dir %{_prefix}/lib/systemd/logind.conf.d
+%dir %{_prefix}/lib/systemd/system-sleep
+%dir %{_sysconfdir}/acpi
+%dir %{_sysconfdir}/acpi/events
+%dir %{_sysconfdir}/acpi/actions
+
+
+%dir %{_sysconfdir}/tuned/nothp_profile
+%config(noreplace) %{_sysconfdir}/tuned/nothp_profile/tuned.conf
+
+%{_prefix}/lib/systemd/system-sleep/sleep-handler.sh
+%{_prefix}/lib/systemd/logind.conf.d/00-hibinit-agent.conf
+
+
+%pre
+%service_add_pre hibinit-agent.service
+
+%post
+%service_add_post hibinit-agent.service
+
+#
+# Disable THP by switching to nothp_profile profile
+#
+tuned-adm profile nothp_profile
```

Comment:
we don't need this anymore? Has SUSE merged the kernel patch? https://lore.kernel.org/linux-mm/20200819195613.24269-1-hsiangkao@redhat.com/